sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/aad/AadResourceServerWebSecurityConfigurerAdapter.java

```diff
@@ -45,8 +70,22 @@ private Converter<Jwt, AbstractAuthenticationToken> jwtAuthenticationConverter()
         if (StringUtils.hasText(properties.getPrincipalClaimName())) {
             converter.setPrincipalClaimName(properties.getPrincipalClaimName());
         }
-        converter.setJwtGrantedAuthoritiesConverter(
-            new AadJwtGrantedAuthoritiesConverter(properties.getClaimToAuthorityPrefixMap()));
+
+        this.jwtGrantedAuthorityConverter = jwtGrantedAuthorityConverter();
+        if (this.jwtGrantedAuthorityConverter != null) {
+            converter.setJwtGrantedAuthoritiesConverter(jwtGrantedAuthorityConverter);
+        } else {
+            converter.setJwtGrantedAuthoritiesConverter(
+                new AadJwtGrantedAuthoritiesConverter(properties.getClaimToAuthorityPrefixMap()));
```

Comment:
this is a little wired, why this `new AadJwtGrantedAuthoritiesConverter(properties.getClaimToAuthorityPrefixMap())` here instead of in the constructor?

Reply:
It's required a Map to construct the granted authorities converter, the `AadJwtGrantedAuthoritiesConverter` only accepts a constructor parameter.